superset/connectors/druid/models.py

```diff
@@ -1237,15 +1237,21 @@ def query(self, query_obj):
 
         # Reordering columns
         cols = []
+        groupby = query_obj.get('groupby') or []
+        columns = query_obj.get('columns') or []
+        metrics = query_obj.get('metrics') or []
         if DTTM_ALIAS in df.columns:
             cols += [DTTM_ALIAS]
-        cols += query_obj.get('groupby') or []
-        cols += query_obj.get('columns') or []
-        cols += query_obj.get('metrics') or []
+        cols += groupby
```

Comment:
cols = [DTTM_ALIAS] + groupby + columns + metrics

may be a bit cheaper and clean